accepted/2025/nuget-mcp.md

```diff
@@ -0,0 +1,265 @@
+
+The sample JSON above leverages this new single-shot command execution.
+
+RID-specific tools solve the "giant package" problem but are not necessarily a hard blocker for the experience.
+
+### Add support for NuGet packages in the MCP registry
+
+#### Today's experience
+
+The [official MCP registry](https://github.com/modelcontextprotocol/registry/discussions/11) is not yet live, but the discussion and protocol description around it clearly list [several supported underlying registries](https://github.com/modelcontextprotocol/registry/blob/a4cefcf05f81466ad65e7c3971e76d0f6d60783e/docs/openapi.yaml#L173-L176) for hosting the local server's code.
+
+Current list: `npm`, `docker`, `pypi`, `homebrew`
+
+#### Add NuGet to the list
+
+We will work with the MCP server registry team to add `nuget` to the list.
+
+We will publish/link to guidance on how to publish a NuGet-based MCP server to the MCP registry. This step is performed after the package is published to NuGet.org.
+
+### Add support for NuGet packages in VS Code
+
+#### Today's experience
+
+Similar to the previous section, NuGet is not a recognized MCP server host.
+
+![VS Code MCP server installation](../../meta/resources/nuget-mcp/vscode-today.png)
+
+#### Add NuGet to the list
+
+We will work with the VS Code team to add a NuGet option to the list, which updates the user's `mcp.json` to invoke `dnx` with the provided package ID.
+
+This is tracked by [microsoft/vscode-copilot#15329](https://github.com/microsoft/vscode-copilot/issues/15329).
+
+Once Visual Studio has a corresponding experience ([the current experience is manually editing the `mcp.json`](https://learn.microsoft.com/en-us/visualstudio/ide/mcp-servers?view=vs-2022)), we will ensure NuGet is supported similarly.
+
+## Rollout plan
+
+1. Enable `McpServer` support on NuGet.org search and the package details page, behind a feature flag.
+2. Work with Microsoft and community MCP server implementers to package as NuGet.
+   - As of June 2025, there are 28 .NET tools on NuGet.org that use the MCP SDK. Newer versions should have the `McpServer` package type.
+   - Some Microsoft MCP servers are implemented in .NET but distributed via npm, for example [@azure/mcp](https://www.npmjs.com/package/@azure/mcp).
+3. Wait for `dnx` to land in a .NET 10 preview.
+4. Work with the MCP registry and VS Code to get NuGet support, perhaps providing PRs/OSS contributions ourselves.
+5. Enable MCP server UI on NuGet.org.
+6. Publish a document on learn.microsoft.com on how to create your own NuGet MCP server. 
+
+## Future Possibilities
+
+We will wait to publish an MCP server template until the .NET MCP SDK has announced a stable API surface area. It is currently in prerelease. In addition we will continue to keep eyes on the development of the [MCP specification](https://modelcontextprotocol.io/development/updates) and [MCP Registry](https://github.com/modelcontextprotocol/registry).
```

Comment:
MCP may be turbulent for quite some time. But MCP support in clients isn't slowing down. Why wait to expose this for a stable .NET MCP SDK? Changes to the MCP SDK won't break servers, since they ship with all their dependencies of a version that they require.

Reply:
This would need agreement with the .NET MCP SDK team. If we fix to a specific version (not floating in the template) and release new template versions frequently, it seems like it could work.